Bio/SeqFeature.py

```diff
@@ -349,8 +349,23 @@ def extract(self, parent_sequence):
         >>> f.extract(seq)
         Seq('VALIVIC', ProteinAlphabet())
 
+        If the FeatureLocation is None, e.g. when parsing invalid locus
+        locations in the GenBank parser, extract() will raise a ValueError.
+
+        >>> from Bio.Seq import Seq
+        >>> from Bio.SeqFeature import SeqFeature
+        >>> seq = Seq("MKQHKAMIVALIVICITAVVAAL", generic_protein)
+        >>> f = SeqFeature(None, type="domain")
+        >>> f.extract(seq)
+        Traceback (most recent call last):
+           ...
+        ValueError: FeatureLocation is None. Check the sequence file for a valid location.
+
         Note - currently only sub-features of type "join" are supported.
         """
+        if self.location is None:
+            raise ValueError("FeatureLocation is None. Check the sequence "
```

Comment:
``FeatureLocation`` is a little misleading (as could also be a ``FeatureLocation``), maybe ``The feature's .location is None.``?